net/sync/service/src/main/java/de/danoeh/antennapod/net/sync/service/SynchronizationQueueStorage.java

```diff
@@ -109,6 +114,39 @@ protected void enqueueFeedAdded(String downloadUrl) {
         }
     }
 
+    /** Remove feed entries that conflict with the given list of current local subscriptions.
+     * <>
+     * This is only relevant for old clients that have legacy data. In newer versions, `enqueueFeedAdded`
+     * and `enqueueFeedAdded` already take care of removing conflicting entries.
+     * */
+    protected void removeLegacyConflictingFeedEntries(Collection<String> currentLocalSubscriptions) {
+
+        List<String> removedQueue = this.getQueuedRemovedFeeds();
+        List<String> addedQueue = this.getQueuedAddedFeeds();
```

Comment:
I think I found a much shorter way to write this function that should give an equivalent result. Did I miss something?

```
removedQueue.removeAll(currentLocalSubscriptions);
addedQueue.removeAll(removedQueue);
```